tests/test_meshtastic_utils_connect_paths.py

```
@@ -357,6 +359,420 @@ def test_connect_meshtastic_logs_firmware_version_on_success(
+
+    startup_deadline_seen_during_metadata: float | None = None
+    startup_deadline_seen_at_subscribe: float | None = None
+
+    monotonic_values = [1_000.0, 1_030.0]
+
+    def _monotonic_side_effect() -> float:
+        if monotonic_values:
+            return monotonic_values.pop(0)
+        return 1_030.0
+
+    def _metadata_side_effect(_client: Any) -> dict[str, Any]:
```

Ruff (0.15.7)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `_client`

(ANN401)